### ISSUE TITLE
Training with GPU

### ISSUE THREAD

**Issue by zhangxiangnick** [Calamari-OCR/calamari#8]
I tried to training a new model on GPU by
```
calamari-train --files data/*.png --batch_size=64
```

I installed calamari-ocr using pip. Got the following error
```
Found 181 files in the dataset
Loading Dataset: 100%|██████████████████████████████████████████████████████████████████████████████████████| 181/181 [00:00<00:00, 242.80it/s]
Text Preprocessing: 100%|█████████████████████████████████████████████████████████████████████████████████| 181/181 [00:00<00:00, 14412.59it/s]
Data Preprocessing: 100%|████████████████████████████████████████████████████████████████████████████████████| 181/181 [00:03<00:00, 56.37it/s]
CODEC: ['', ' ', "'", '-', '.', '1', '2', 'ء', 'آ', 'أ', 'ؤ', 'ئ', 'ا', 'ب', 'ة', 'ت', 'ث', 'ج', 'ح', 'خ', 'د', 'ذ', 'ر', 'ز', 'س', 'ش', 'ص', 'ض', 'ط', 'ظ', 'ع', 'غ', 'ف', 'ق', 'ك', 'ل', 'م', 'ن', 'ه', 'و', 'ى', 'ي', 'ً', 'َ']
/home/ubuntu/anaconda3/envs/python3/lib/python3.6/importlib/_bootstrap.py:219: RuntimeWarning: numpy.dtype size changed, may indicate binary incompatibility. Expected 96, got 88
  return f(*args, **kwds)
2018-08-08 05:28:36.538017: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2018-08-08 05:28:36.622038: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:897] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
2018-08-08 05:28:36.622469: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1392] Found device 0 with properties: 
name: Tesla K80 major: 3 minor: 7 memoryClockRate(GHz): 0.8235
pciBusID: 0000:00:1e.0
totalMemory: 11.17GiB freeMemory: 11.10GiB
2018-08-08 05:28:36.622503: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1471] Adding visible gpu devices: 0
2018-08-08 05:28:36.912184: I tensorflow/core/common_runtime/gpu/gpu_device.cc:952] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-08-08 05:28:36.912241: I tensorflow/core/common_runtime/gpu/gpu_device.cc:958]      0 
2018-08-08 05:28:36.912255: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971] 0:   N 
2018-08-08 05:28:36.912544: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1084] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10761 MB memory) -> physical GPU (device: 0, name: Tesla K80, pci bus id: 0000:00:1e.0, compute capability: 3.7)
Using CUDNN LSTM backend on GPU
Using CUDNN LSTM backend on GPU
Traceback (most recent call last):
  File "/home/ubuntu/anaconda3/envs/python3/bin/calamari-train", line 11, in <module>
    sys.exit(main())
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/calamari_ocr/scripts/train.py", line 233, in main
    run(args)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/calamari_ocr/scripts/train.py", line 226, in run
    trainer.train(progress_bar=not args.no_progress_bars)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/calamari_ocr/ocr/trainer.py", line 163, in train
    test_net = backend.create_net(restore=None, weights=self.weights, graph_type="test", batch_size=checkpoint_params.batch_size)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/calamari_ocr/ocr/backends/tensorflow_backend/tensorflow_backend.py", line 26, in create_net
    model = TensorflowModel(self.network_proto, self.graph, self.session, graph_type, batch_size, reuse_weights=not self.first_model)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/calamari_ocr/ocr/backends/tensorflow_backend/tensorflow_model.py", line 44, in __init__
    self.create_network(self.inputs, self.input_seq_len, self.dropout_rate, reuse_variables=reuse_weights)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/calamari_ocr/ocr/backends/tensorflow_backend/tensorflow_model.py", line 162, in create_network
    time_major_outputs = gpu_cudnn_lstm_backend(time_major_inputs, lstm_layers[0].hidden_nodes)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/calamari_ocr/ocr/backends/tensorflow_backend/tensorflow_model.py", line 156, in gpu_cudnn_lstm_backend
    time_major_outputs, (output_h, output_c) = rnn_lstm(time_major_inputs)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/tensorflow/python/layers/base.py", line 329, in __call__
    outputs = super(Layer, self).__call__(inputs, *args, **kwargs)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py", line 696, in __call__
    self.build(input_shapes)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/tensorflow/contrib/cudnn_rnn/python/layers/cudnn_rnn.py", line 362, in build
    initializer=opaque_params_t, validate_shape=False)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 1328, in get_variable
    constraint=constraint)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 1090, in get_variable
    constraint=constraint)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 427, in get_variable
    return custom_getter(**custom_getter_kwargs)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/tensorflow/contrib/cudnn_rnn/python/layers/cudnn_rnn.py", line 294, in _update_trainable_weights
    variable = getter(*args, **kwargs)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 404, in _true_getter
    use_resource=use_resource, constraint=constraint)
  File "/home/ubuntu/anaconda3/envs/python3/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 761, in _get_single_variable
    "reuse=tf.AUTO_REUSE in VarScope?" % name)
ValueError: Variable cudnn_lstm_1/opaque_kernel does not exist, or was not created with tf.get_variable(). Did you mean to set reuse=tf.AUTO_REUSE in VarScope?
```

So it seems it find the GPU device, but got problem in 
```
time_major_outputs, (output_h, output_c) = rnn_lstm(time_major_inputs)
```

Thanks!

**Comment by zhangxiangnick** [comment]
Okay I think the problem is with TensorFlow 1.9.0. This is what pip installed by default. After I downgrade it to 1.8.0, the training works.

**Comment by ChWick** [comment]
Thanks for the info! I'll consider an update to tensorflow 1.9 in the next few days!

**Comment by zhangxiangnick** [comment]
@erjinzhi2333 in my case, pip install tensorflow-gpu==1.8.

**Comment by mbaghert** [comment]
After Downgrade to tensorflow-gpu==1.8


InvalidArgumentError (see above for traceback): Cannot colocate nodes 'global_norm_1/L2Loss_4' and 'gradients/cudnn_lstm/CudnnRNN_grad/CudnnRNNBackprop' because no device type supports both of those nodes and the other nodes colocated with them.
Colocation Debug Info:
Colocation group had the following types and devices:
CudnnRNNBackprop: GPU
Identity:
L2Loss: CPU

**Comment by ChWick** [comment]
@mbaghert Can you please append your call, operating system, Cuda and Cudnn version, this might help to debug you issue. Maybe create a new issue on github for this specific issue, too.

**Comment by mbaghert** [comment]
Windows 10
CUDA v9.0
Cudnn 7

**Comment by mbaghert** [comment]
Collecting tensorboard<1.9.0,>=1.8.0 (from tensorflow-gpu==1.8)
  Using cached https://files.pythonhosted.org/packages/59/a6/0ae6092b7542cfedba6b2a1c9b8dceaf278238c39484f3ba03b03f07803c/tensorboard-1.8.0-py3-none-any.whl
Requirement already satisfied: numpy>=1.13.3 in c:\programdata\anaconda3\lib\site-packages (from tensorflow-gpu==1.8) (1.14.5)
Requirement already satisfied: grpcio>=1.8.6 in c:\programdata\anaconda3\lib\site-packages (from tensorflow-gpu==1.8) (1.15.0)
Requirement already satisfied: absl-py>=0.1.6 in c:\programdata\anaconda3\lib\site-packages (from tensorflow-gpu==1.8) (0.4.1)
Requirement already satisfied: gast>=0.2.0 in c:\programdata\anaconda3\lib\site-packages (from tensorflow-gpu==1.8) (0.2.0)
Requirement already satisfied: six>=1.10.0 in c:\programdata\anaconda3\lib\site-packages (from tensorflow-gpu==1.8) (1.11.0)
Requirement already satisfied: termcolor>=1.1.0 in c:\programdata\anaconda3\lib\site-packages (from tensorflow-gpu==1.8) (1.1.0)
Requirement already satisfied: astor>=0.6.0 in c:\programdata\anaconda3\lib\site-packages (from tensorflow-gpu==1.8) (0.7.1)
Requirement already satisfied: protobuf>=3.4.0 in c:\programdata\anaconda3\lib\site-packages (from tensorflow-gpu==1.8) (3.6.1)
Requirement already satisfied: html5lib==0.9999999 in c:\programdata\anaconda3\lib\site-packages (from tensorboard<1.9.0,>=1.8.0->tensorflow-gpu==1.8) (0.9999999)
Requirement already satisfied: werkzeug>=0.11.10 in c:\programdata\anaconda3\lib\site-packages (from tensorboard<1.9.0,>=1.8.0->tensorflow-gpu==1.8) (0.14.1)
Requirement already satisfied: bleach==1.5.0 in c:\programdata\anaconda3\lib\site-packages (from tensorboard<1.9.0,>=1.8.0->tensorflow-gpu==1.8) (1.5.0)
Requirement already satisfied: markdown>=2.6.8 in c:\programdata\anaconda3\lib\site-packages (from tensorboard<1.9.0,>=1.8.0->tensorflow-gpu==1.8) (2.6.11)
Requirement already satisfied: setuptools in c:\programdata\anaconda3\lib\site-packages (from protobuf>=3.4.0->tensorflow-gpu==1.8) (39.1.0)

**Comment by mbaghert** [comment]
problem solved by conda install tensorflow-gpu==1.8
